src/app_modes/image_view.rs

```diff
@@ -0,0 +1,122 @@
+    pub fn new(image_topics: Vec<ListenerConfig>) -> ImageView {
+        let mut images: Vec<image::ImageListener> = Vec::new();
+        for image_config in image_topics {
+            images.push(image::ImageListener::new(image_config));
+        }
+        ImageView {
+            images: images,
+            active_sub: 0,
+        }
+    }
+}
+
+impl AppMode for ImageView {
+    fn run(&mut self) {
+        if self.images.len() > 0 && !self.images[self.active_sub].is_active() {
+            self.images[self.active_sub].activate();
+        }
+    }
+
+    fn reset(&mut self) {
+        for sub in self.images.iter_mut() {
+            if sub.is_active() {
+                sub.deactivate();
+            }
+        }
+    }
+
+    fn handle_input(&mut self, input: &String) {
+        if self.images.len() > 0 {
+            match input.as_str() {
+                input::LEFT => {
+                    self.images[self.active_sub].deactivate();
+                    self.active_sub = if self.active_sub > 0 {
+                        self.active_sub - 1
+                    } else {
+                        self.images.len() - 1
+                    };
+                }
+                input::RIGHT => {
+                    self.images[self.active_sub].deactivate();
+                    self.active_sub = (self.active_sub + 1) % self.images.len();
+                }
+                _ => (),
+            }
+        }
+    }
+
+    fn get_description(&self) -> Vec<String> {
+        vec!["This mode allows to visualized images received on the given topics.".to_string()]
+    }
+
+    fn get_keymap(&self) -> Vec<[String; 2]> {
+        vec![
+            [
+                input::LEFT.to_string(),
+                "Switches to the previous image.".to_string(),
+            ],
+            [
+                input::RIGHT.to_string(),
+                "Switches to the next image.".to_string(),
+            ],
+        ]
+    }
+
+    fn get_name(&self) -> String {
+        "Image".to_string()
+    }
+}
+
+impl<B: Backend> Drawable<B> for ImageView {
+    fn draw(&self, f: &mut Frame<B>) {
+        let chunks = Layout::default()
+            .constraints([Constraint::Length(1), Constraint::Percentage(100)].as_ref())
+            .split(f.size());
+        if self.images.len() == 0 {
+            let header = Paragraph::new(Spans::from(Span::raw(
+                self.get_name() + " view - No topic configured!",
+            )))
+            .block(Block::default().borders(Borders::NONE))
+            .style(Style::default().fg(Color::White))
+            .alignment(Alignment::Center)
+            .wrap(Wrap { trim: false });
+            f.render_widget(header, chunks[0]);
+        } else {
+            for image_sub in &self.images {
+                if image_sub.is_active() {
+                    let header = Paragraph::new(Spans::from(Span::raw(
+                        self.get_name() + " view - Topic: /" + &image_sub.config.topic,
```

Comment:
I got a bit distracted from the long sentence. How about the app mode name aligned left and additional information aligned right?

Reply:
Turns out in the title of a canvas it is not possible (or I haven't found a way), so I made the title red and bold and the info normal. For coherency I made the image mode the same